v2/internal/typescriptify/typescriptify.go

```diff
@@ -261,15 +275,22 @@ func (t *TypeScriptify) AddType(typeOf reflect.Type) *TypeScriptify {
 func (t *typeScriptClassBuilder) AddMapField(fieldName string, field reflect.StructField) {
 	keyType := field.Type.Key()
 	valueType := field.Type.Elem()
-	valueTypeName := valueType.Name()
+	valueTypeName := nameTypeOf(valueType)
+	if valueType.Kind() == reflect.Ptr {
+		valueType = valueType.Elem()
+	}
 	if name, ok := t.types[valueType.Kind()]; ok {
 		valueTypeName = name
 	}
 	if valueType.Kind() == reflect.Array || valueType.Kind() == reflect.Slice {
-		valueTypeName = valueType.Elem().Name() + "[]"
+		valueTypeName = nameTypeOf(valueType.Elem()) + "[]"
 	}
 	if valueType.Kind() == reflect.Ptr {
-		valueTypeName = valueType.Elem().Name()
+		valueTypeName = nameTypeOf(valueType.Elem())
+	}
+	if valueType.Kind() == reflect.Map {
```

Comment:
previously there was no reflect.Map here, so valueTypeName defaulted to "" and caused invalid typescript with e.g. nested map[string]map[string]string.

any here gives valid code at least, future work to recursively generate?